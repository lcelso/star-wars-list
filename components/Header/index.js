import React from 'react';
import styled from 'styled-components';
import Link from 'next/link';

const HeaderElment = styled.header`
  background-color: #040916;
  height: 60px;
  padding: 20px;
`

const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(1, 1fr);
  grid-column-gap: 5px;
  grid-row-gap: 20px;
`;

const ItemLeft = styled.div`
  text-align: left;
`

const ItemCenter = styled.div`
  text-align: center;
`

const ItemRight = styled.div`
  text-align: right;
`

const Points = styled.ul`
  display: flex;
  justify-content: center;
  align-self: center;
`
const PointItems = styled.span`
  background-color: #FFFFFF;
  padding: 12px;
  color: #000000;
  display: inline-block;
`;

const Header = ({ total }) => {
  const { error, correct, points } = total;

  return(
    <HeaderElment>
      <Grid>
        <ItemLeft>
          <Link href="/">
            <a alt="Voltar para home"><img src="/images/logo.png" width="130" alt="Star Wars" /></a>
          </Link>
        </ItemLeft>
        <ItemCenter>
          <p>Vamos descobrir mais sobre Star Wars?</p>
          <p>Neste game vocÃª tera dicas para descobrir cada personagem. Veja abaixo</p>
        </ItemCenter>
        <ItemRight>
          <Points>
            <li><img src="/images/icon_points.png" width="40" /></li>
            <li><PointItems>{points} Pontos</PointItems></li>
            <li><img src="/images/icon_correct.png" width="40" /></li>
            <li><PointItems>{correct} Acertos</PointItems></li>
            <li><img src="/images/icon_error.png" width="40" /></li>
            <li><PointItems>{error} Erros</PointItems></li>
          </Points>
        </ItemRight>
      </Grid>
    </HeaderElment>
  );
};

export default Header;
