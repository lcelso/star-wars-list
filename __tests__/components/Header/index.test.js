import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from '../../../components/Header';

describe('Component - Loading', () => {
  it('should render correctly in "debug" mode', () => {
    const data = { error: 0, correct: 0, points: 0 }
    const component = shallow(<Header total={data} />);
  
    expect(component).toMatchSnapshot();
  });

  it('it must be possible to access the html element of the link to return', () => {
    const data = { error: 0, correct: 0, points: 0 }
    const component = mount(<Header total={data} />);
    const mockElement = '<a id="btn-back" alt="Voltar para home" href="/"><img src="/images/logo.png" width="130" alt="Star Wars"></a>'

    expect(component.find('#btn-back').html()).toEqual(mockElement);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});