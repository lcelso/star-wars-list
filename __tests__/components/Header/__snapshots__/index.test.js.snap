// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Loading it must be possible to access the html element of the link to return 1`] = `ReactWrapper {}`;

exports[`Component - Loading should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
